src/CoreBundle/bin/console

```diff
@@ -1,23 +1,36 @@
 #!/usr/bin/php
 <?php
 
+use App\Kernel;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Debug\Debug;
+
 set_time_limit(0);
 
+require __DIR__.'/../../../../../../vendor/autoload.php';
+
 /**
  * set $_SERVER vars manually, as they are not available in console.
  */
-$_SERVER['DOCUMENT_ROOT'] = realpath(__DIR__.'/../../../../../../web');
+$_SERVER['DOCUMENT_ROOT'] = realpath(__DIR__.'/../../../../../../public');
 $_SERVER['HTTP_HOST'] = 'you-have-no-host-because-you-are-in-console.com';
 define('REQUEST_PROTOCOL', 'console');
 
 define('PATH_WEB', $_SERVER['DOCUMENT_ROOT']);
 require_once __DIR__.'/../Resources/config/const.inc.php';
 
-if (file_exists(PATH_PROJECT_BASE.'/app/autoload.php')) {
-    include_once PATH_PROJECT_BASE.'/app/autoload.php';
-} else {
-    include_once __DIR__.'/../../../../../autoload.php';
+$input = new ArgvInput();
+if (null !== $_ENV['APP_ENV'] = $input->getParameterOption(['--env', '-e'], null, true)) {
+    putenv('APP_ENV='.$_ENV['APP_ENV']);
+    // force loading .env files when --env is defined
+    $_SERVER['APP_ENV'] = null;
+}
```

Comment:
This whole block is a little confusing. First we set a value inside $_ENV by hand, then we transfer it to the outer environment - where it should have come from in the first place - and after that, we set a value in $_SERVER with the same name to `null`. I am sure this produces certain side effects we want at this point, but this code is not obvious about that at all. If we need to do it this way, a small doc paragraph explaining the thoughts behind it would be neat.